app/controllers/admin/enterprises_controller.rb

```diff
@@ -114,8 +114,36 @@ def visible
       end
     end
 
+    def invite_manager
+      existing_user = Spree::User.where("email = :email OR unconfirmed_email = :email", email: params[:email]).first
+
+      if existing_user
+        render json: { errors: t('admin.enterprises.invite_manager.user_already_exists') }, status: :unprocessable_entity
+        return
+      end
+
+      new_user = create_new_manager(params[:email], params[:enterprise])
+
+      if new_user
+        render json: { user: new_user.id }, status: :ok
+      else
+        render json: { errors: t('admin.enterprises.invite_manager.error') }, status: 500
+      end
+    end
+
     protected
 
+    def create_new_manager(email, enterprise_id)
```

Comment:
following the Rest convention, this could become a `save` method inside a `Invitation` class which wouldn't add yet another responsibility to this already big controller.